### PR TITLE
Ledger model section on transaction normalization.

### PR DESCRIPTION
**Pull request by sofiafaro-da**
The main reason to bring normalization out as a separate section (rather have it be part of the projection section) is to be able to say that all transactions must be normalized, and that Daml code only produces normalized transactions. This also makes the section on projection much simpler (and projection serves as a practical example for normalization).

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/concepts/ledger-model/ledger-exceptions.rst [hunk 1]
```diff
@@ -122,6 +122,84 @@ And the third:
 
 As you can see, in each of these continuities, no contract was consumed twice.
 
+Transaction Normalization
++++++++++++++++++++++++++
+
+The same "before-after" relation can be represented in more than one way using
+rollback nodes. For example, the following three transactions have the same
+"before-after" relation among their ledger actions (A and B):
```
Comment (andreaslochbihler-da) [inline docs/source/concepts/ledger-model/ledger-exceptions.rst:130]: Elsewhere in the ledger model, we use upper-case letters `A` and `B` to represent parties. Actions are typically called `act` with indices.
```suggestion
"before-after" relation among their ledger actions (`act1` and `act2`):
```

File: docs/source/concepts/ledger-model/ledger-exceptions.rst [hunk 2]
```diff
@@ -122,6 +122,84 @@ And the third:
...
+**Transaction normalization** is the process by which equivalent transactions
+are converted into the same transaction. In the case above, all three
+transactions become the transaction in the middle when normalized.
+
+.. https://lucid.app/lucidchart/8a5a09a1-5473-4abf-a72a-57bf03b56794/edit
+.. image:: ./images/exception-normalization-2.svg
+   :align: center
+   :width: 80%
+
+To normalize a transaction, we apply three rules across the whole transaction:
```
Comment (andreaslochbihler-da) [inline docs/source/concepts/ledger-model/ledger-exceptions.rst:164]: Minor: Do you want to walk the reader through the normalization of the diagrams? It's not plain obvious how to get from the left to the middle (and there are actually two ways to get there: use "pull out the start" or "pull out the end" and then remove the empty remaining rollback node.

**Review by andreaslochbihler-da — APPROVED** [review]
Looks good. Left a few small comments.

File: docs/source/concepts/ledger-model/ledger-exceptions.rst [hunk 3]
```diff
@@ -122,6 +122,84 @@ And the third:
...
+   :align: center
+   :width: 80%
+
+Because of this, these three transactions are equivalent.
+More generally, two transactions are equivalent if:
+
+- The transactions are the same when you ignore all rollback nodes. That is,
+  if you remove every rollback node and absorb its children into its parent,
+  then two transactions are the same. Equivalently, the transactions have
+  the same ledger actions with the same prefix order and subaction relation.
```
Comment (cocreature) [inline docs/source/concepts/ledger-model/ledger-exceptions.rst:143]: do you mean prefix order or the same preorder traversal?

**Review by cocreature — APPROVED** [review]
Nice, thank you!

File: docs/source/concepts/ledger-model/ledger-exceptions.rst (hunk 3 above)
Comment (sofiafaro-da) [reply docs/source/concepts/ledger-model/ledger-exceptions.rst:143]: Ah, preorder traversal makes more sense :-)

File: docs/source/concepts/ledger-model/ledger-exceptions.rst (hunk 2 above)
Comment (sofiafaro-da) [reply docs/source/concepts/ledger-model/ledger-exceptions.rst:164]: Good idea. I changed the example a bit to eliminate ambiguity and to showcase both rules 2 and 3.

**Pull request merged**